operator/controllers/actionsrunner/reconciler.go

```diff
@@ -0,0 +1,298 @@
+	if err := r.Get(ctx, req.NamespacedName, &actionsRunner); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	var actionsRunnerJob inlocov1alpha1.ActionsRunnerJob
+	switch err := r.Get(ctx, req.NamespacedName, &actionsRunnerJob); {
+	case err == nil:
+		key := fmt.Sprintf("%s/%s", actionsRunnerJob.GetNamespace(), actionsRunnerJob.GetName())
+		if _, loaded := r.waiting.LoadOrStore(key, nil); !loaded {
+			go r.waitForJob(ctx, &actionsRunnerJob, nil)
+		}
+		return ctrl.Result{}, nil
+	case !apierrors.IsNotFound(err):
+		return ctrl.Result{}, err
+	}
+
+	var configMap corev1.ConfigMap
+	if err := r.Get(ctx, req.NamespacedName, &configMap); client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+
+	var secret corev1.Secret
+	if err := r.Get(ctx, req.NamespacedName, &secret); client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+
+	wire, ok := r.wires[req.NamespacedName]
+	if !ok {
+		wire = &Wire{
+			Context:       ctx,
+			Log:           log,
+			ActionsRunner: &actionsRunner,
+			DotFiles:      toDotFiles(&configMap, &secret),
+		}
+
+		log.Info("Initializing Wire")
+
+		if err := wire.Init(); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		log.Info("Wire Initialized")
+
+		r.wires[req.NamespacedName] = wire
+	}
+
+	desiredConfigMap, err := toConfigMap(wire.DotFiles, &actionsRunner, r.Scheme)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	desiredSecret, err := toSecret(wire.DotFiles, &actionsRunner, r.Scheme)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.Patch(ctx, desiredConfigMap, client.Apply, patchOpts...); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.Patch(ctx, desiredSecret, client.Apply, patchOpts...); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.consume(wire); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.source = make(chan event.GenericEvent)
+	r.wires = make(map[types.NamespacedName]*Wire)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&inlocov1alpha1.ActionsRunner{}).
+		Owns(&inlocov1alpha1.ActionsRunnerJob{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.PersistentVolumeClaim{}).
+		Owns(&batchv1.Job{}).
+		Watches(&source.Channel{Source: r.source}, &handler.EnqueueRequestForObject{}).
+		Complete(r)
+}
+
+func (r *Reconciler) consume(wire *Wire) error {
+	ack, msg, err := wire.Go(r.source)
+	select {
+	case msg := <-msg:
+		switch typ := msg.Type; typ {
+		case facades.MessageTypePipelineAgentJobRequest:
+			return r.onPipelineAgentJobRequest(wire, ack, msg)
+		case facades.MessageTypeJobCancellation:
+			return r.onPipelineAgentJobRequest(wire, ack, msg)
+		case facades.MessageTypeAgentRefresh:
+			return r.onPipelineAgentJobRequest(wire, ack, msg)
+		default:
+			return fmt.Errorf("unknown MessageType(%s)", typ)
+		}
+	case err := <-err:
+		return err
+	default:
```

Comment:
I would recommend writing a comment here just to make sure the reader understands that this default case is intended

Reply:
ok